### ISSUE TITLE
salt minion won't start: Non valid IP address match on BSD alias format

### ISSUE THREAD

**Issue by iMilnb** [saltstack/salt#29833]
While starting a minion on NetBSD with a couple of IP aliases (will probably happen on FreeBSD and OpenBSD in this scenario) I get the following traceback:
```
2015-12-18 11:03:40,410 [salt.scripts                             ][ERROR   ][18580] Minion failed to start: 
Traceback (most recent call last):
  File "/usr/pkg/lib/python2.7/site-packages/salt/scripts.py", line 81, in minion_process
    minion.start()
  File "/usr/pkg/lib/python2.7/site-packages/salt/cli/daemons.py", line 272, in start
    self.prepare()
  File "/usr/pkg/lib/python2.7/site-packages/salt/cli/daemons.py", line 254, in prepare
    self.minion = salt.minion.Minion(self.config)
  File "/usr/pkg/lib/python2.7/site-packages/salt/minion.py", line 708, in __init__
    self.opts['grains'] = salt.loader.grains(opts)
  File "/usr/pkg/lib/python2.7/site-packages/salt/loader.py", line 613, in grains
    ret = fun()
  File "/usr/pkg/lib/python2.7/site-packages/salt/grains/core.py", line 1430, in ip4
    return {'ipv4': salt.utils.network.ip_addrs(include_loopback=True)}
  File "/usr/pkg/lib/python2.7/site-packages/salt/utils/network.py", line 993, in ip_addrs
    return _ip_addrs(interface, include_loopback, interface_data, 'inet')
  File "/usr/pkg/lib/python2.7/site-packages/salt/utils/network.py", line 981, in _ip_addrs
    addr = ipaddress.ip_address(addr.get('address'))
  File "/usr/pkg/lib/python2.7/site-packages/salt/ext/ipaddress.py", line 117, in ip_address
    address)
ValueError: 'alias' does not appear to be an IPv4 or IPv6 address
2015-12-18 11:03:40,420 [salt.scripts                             ][WARNING ][18580] ** Restarting minion **
```
This error probably comes from the fact that BSD IP aliases are shown this way:
```
xennet0: flags=8863<UP,BROADCAST,NOTRAILERS,RUNNING,SIMPLEX,MULTICAST> mtu 1500
        capabilities=2800<TCP4CSUM_Tx,UDP4CSUM_Tx>
        enabled=0
        address: 00:50:57:19:e0:a0
        inet 192.168.0.2 netmask 0xffffff00 broadcast 192.168.0.255
        inet alias 172.16.0.1 netmask 0xffffffff broadcast 172.16.0.1
[...]
```

**Comment by iMilnb** [comment]
Simple fix:
in `salt/utils/network.py`:
replace: `pip = re.compile(r'.*?(?:inet addr:|inet )(.*?)\s')`
with: `pip = re.compile(r'.*?(?:inet addr:|inet[^0-9]+)(.*?)\s')`

**Comment by jfindlay** [comment]
@iMilnb, thanks for the report.  What version of salt are you using?

**Comment by iMilnb** [comment]
@jfindlay salt 2015.8.3

**Comment by cachedout** [comment]
I'm not a FreeBSD user, so forgive my ignorance, but I'm not seeing that same output from `ifconfig` that you seem to be showing. After I add some aliases like follows:

`root@mp-bsd:~ # ifconfig xn1 127.0.0.129 netmask 255.255.255.255 alias`

I see ifconfig output like this:

```
root@mp-bsd:~ # ifconfig -a
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> metric 0 mtu 16384
	options=600003<RXCSUM,TXCSUM,RXCSUM_IPV6,TXCSUM_IPV6>
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	inet 127.0.0.1 netmask 0xff000000 
	nd6 options=23<PERFORMNUD,ACCEPT_RTADV,AUTO_LINKLOCAL>
xn0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=503<RXCSUM,TXCSUM,TSO4,LRO>
	ether bc:76:4e:11:d4:2b
	inet6 fe80::be76:4eff:fe11:d42b%xn0 prefixlen 64 scopeid 0x2 
	inet 192.237.188.54 netmask 0xffffff00 broadcast 192.237.188.255 
	inet6 2001:4801:7821:77:be76:4eff:fe11:d42b prefixlen 64 
	nd6 options=23<PERFORMNUD,ACCEPT_RTADV,AUTO_LINKLOCAL>
	media: Ethernet manual
	status: active
xn1: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=503<RXCSUM,TXCSUM,TSO4,LRO>
	ether bc:76:4e:11:b2:8f
	inet6 fe80::be76:4eff:fe11:b28f%xn1 prefixlen 64 scopeid 0x3 
	inet 10.208.130.111 netmask 0xffff8000 broadcast 10.208.255.255 
	inet 127.0.0.127 netmask 0xffffffff broadcast 127.0.0.127 
	inet 127.0.0.128 netmask 0xffffffff broadcast 127.0.0.128 
	inet 127.0.0.129 netmask 0xffffffff broadcast 127.0.0.129 
	nd6 options=23<PERFORMNUD,ACCEPT_RTADV,AUTO_LINKLOCAL>
	media: Ethernet manual
	status: active
```

Salt's utility functions seem to have no problem with this:

```
root@mp-bsd:~ # python2
Python 2.7.10 (default, Jul 23 2015, 01:22:23) 
[GCC 4.2.1 Compatible FreeBSD Clang 3.4.1 (tags/RELEASE_34/dot1-final 208032)] on freebsd10
Type "help", "copyright", "credits" or "license" for more information.
>>> import salt.utils.network
>>> salt.utils.network.ip_addrs
<function ip_addrs at 0x807511aa0>
>>> salt.utils.network.ip_addrs()
['10.208.130.111', '192.237.188.54']
>>> salt.utils.network.ip_addrs(include_loopback=True)
['10.208.130.111', '127.0.0.1', '127.0.0.127', '127.0.0.128', '127.0.0.129', '192.237.188.54']
>>> quit
Use quit() or Ctrl-D (i.e. EOF) to exit
>>> 
```

Where am I going wrong here?

**Comment by iMilnb** [comment]
@cachedout seems like FreeBSD does not indicate if an IP address is an alias or not. In NetBSD we have the following:
```
xennet0: flags=8863<UP,BROADCAST,NOTRAILERS,RUNNING,SIMPLEX,MULTICAST> mtu 1500
        capabilities=2800<TCP4CSUM_Tx,UDP4CSUM_Tx>
        enabled=0
        address: 00:50:57:0a:00:04
        inet 192.168.0.1 netmask 0xffffff00 broadcast 192.168.0.255
        inet alias 172.16.0.1 netmask 0xffffffff broadcast 172.16.0.1
        inet alias 172.16.1.1 netmask 0xffffff00 broadcast 172.16.1.255
[...]
```
Obviously FreeBSD != NetBSD. Don't know how OpenBSD shows it.

**Comment by iMilnb** [comment]
Interesting, when calling salt functions I don't reproduce this bug:
```
[~] python2.7
Python 2.7.10 (default, Nov 26 2015, 14:07:58) 
[GCC 4.5.3] on netbsd6
Type "help", "copyright", "credits" or "license" for more information.
>>> import salt.utils.network
>>> salt.utils.network.ip_addrs
<function ip_addrs at 0x7f7ff4fb8a28>
>>> salt.utils.network.ip_addrs()
['10.0.0.1', '172.16.0.1', '172.16.255.1', '172.16.255.2', '192.168.1.2']
>>> salt.utils.network.ip_addrs(include_loopback=True)
['10.0.0.1', '127.0.0.1', '172.16.0.1', '172.16.255.1', '172.16.255.2', '192.168.67.2']
```

**Comment by iMilnb** [comment]
A simple fix consists in replacing:
    pip = re.compile(r'.*?(?:inet addr:|inet )(.*?)\s')
with:
    pip = re.compile(r'.*?(?:inet addr:|inet [^\d]*)(.*?)\s')
in `salt/utils/network.py`

**Comment by cachedout** [comment]
@iMilnb Would you like to send that change over as a PR? We can run the test suite on it and see how things go.

**Comment by cachedout** [comment]
Fixed by #30124. Closing.
